support-frontend/app/controllers/CreateSubscriptionController.scala

```diff
@@ -157,17 +214,21 @@ class CreateSubscriptionController(
         case Some(userDetails) => EitherT.pure[Future, CreateSubscriptionError](userDetails)
         case None =>
           getOrCreateIdentityUser(request.body, request.headers.get("Referer"))
+            .map(userDetails => UserDetailsWithSignedInStatus(userDetails, isSignedIn = false))
             .leftMap(mapIdentityErrorToCreateSubscriptionError)
       }
-      supportWorkersUser = buildSupportWorkersUser(userDetails, request.body, testUsers.isTestUser(request))
+
+      _ <- validateUserIsEligibleForPurchase(request, userDetails)
```

Comment:
Can this validation not be folded into the validate function on line 155 which is already set up to run product specific validation?

Reply:
I've looked into this. I think it could be done, but would require reordering some things. Right now the validation performed by the existing `validate` method doesn't require any information about the current user, so can happen before the `getOrCreateIdentityUser` call. Validating whether the user is eligible for purchasing Ad-Lite requires an identity ID, so has to happen after that call. I guess this keeps that initial `validate` method quite lightweight and fast to fail. We'd have to add `UserDetails` to the signature of `validate`. So keeping them separate seems okay to me, what do you reckon?